osf/models/node.py

```diff
@@ -1797,6 +1784,15 @@ def save(self, *args, **kwargs):
                 self.bulk_update_search(batch)
                 children = children[99:]
 
+        if first_save:
+            self.update_group_permissions()
```

Comment:
Calling update_group_permissions() (GuardianMixin method) on first save to create the three groups and add the appropriate permissions to the group.  Then, I add the creator to the node's admin group.

Creating a contributor is a two-step process, now. You create the Contributor object (which stores visible, order information), and then you call add_permission which adds the user to the appropriate group and gives them permissions.  The `add_contributor` method takes care of both of these steps.  Calling them separately here to avoid some of the other code that's called in that method. (For example, creators don't get an email that they've been added as a contributor).